### PR TITLE
Add documentation for the Clojars release process

### LINKED ISSUE #9: Add docs for the clojars release process
- How to sign up for clojars and deal with credentials
- Adding someone to the group so they can upload releases
- How to cut a release and push changes back to master using `lein release`

### PR DESCRIPTION
**Pull request by ehashman**
Fixes #9.

### CODE REVIEW

File: doc/release.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+4. Assuming you have an up-to-date copy of the cloudpassage-lib git repository
+   checked out, you are now ready to prepare a release.
+
+## Packaging and uploading a release
+
+We use the `lein-release` plugin to release new versions of the library. Its
+docs are a little sparse, so here are some instructions to help you prepare
+releases.
+
```
Comment (derwolfe) [inline doc/release.md]: What would you think of adding another step preceding step step that shows how `~/.lein/profiles.clj` would look with `[lein-release "1.0.9"]` listed in the plugins?

**Comment by derwolfe** [comment]
Nice work!

File: doc/release.md (hunk 1 above)
Comment (ehashman) [inline doc/release.md]: Oh that's a good idea. I forgot about that! Let's do it.

Comment (ehashman) [inline doc/release.md]: I added this above in the first section as step 4 since it's a one-time setup step.

**Comment by derwolfe** [comment]
LGTM :-)

**Pull request merged**